src/models/zindex/zindex.ts

```diff
@@ -0,0 +1,63 @@
+import {action, computed, makeObservable, observable} from "mobx";
+
+export class FloatingObjzIndexManager {
```

Comment:
Just to provide another idea of implementation, no action needed, we could have an array of dialog id arranged in the order of precedence and assign z-index according to the order of the array, instead of a map that stores the dialog id key and z-index value.

Reply:
Indeed, that was the original method of controlling the order of floating widgets. The draggable dialog has a slightly different structure from the floating widget, so I decided to use ID to find the z-index. Anyway, we need to go over the entire z-index array when updating the z-index, I think mapping should be okay.